swaybar/ipc.c

```diff
@@ -298,6 +304,19 @@ bool ipc_get_workspaces(struct swaybar *bar) {
 					calloc(1, sizeof(struct swaybar_workspace));
 				ws->num = json_object_get_int(num);
 				ws->name = strdup(json_object_get_string(name));
+				char *ws_dup = strdup(ws->name);
+				char *stripped_name = strtok(ws_dup, ":");
+				// ws->num will be 0 if workspace name doesn't begin with int.
+				if (stripped_name != NULL && 
+						strlen(ws->name) > 2 && ws->num != 0) {
```

Comment:
Not 0, -1.
Also, using a length of 2 misses cases when num > 9.
I think you should do `strlen(ws->name) > strlen(stripped_name) + 1`. (or set some variable to `strlen(stripped_name) + 1` and use that as the offset later too)

Reply:
According to documentation, `atoi` returns 0 if invalid. Existing code already assumes that `num` is 0-9.

Reply:
https://github.com/swaywm/sway/blob/7fa7f4f48d17e0470c800b258061d188ceb705da/sway/ipc-json.c#L191 gives -1 for num if there isn't a number.

Reply:
Yeah you're right, thanks.